fix(allocator)!: remove `vec!` macro

overlookmotel opened the pull request:
Remove `vec!` macro from `oxc_allocator`. This is in the `vec2::raw_vec` module, but is accidentally exposed due to the `#[macro_export]` attribute.

This macro creates a `bumpalo::collections::Vec`, instead of our own `Vec` type, which is confusing.

We could add a `vec!` macro which creates our own `Vec` type, but that's not completely trivial (how to handle cloning `item` in `vec![item; 3]`? `Clone` or `CloneIn`?). So for now, just remove this erroneous macro entirely.

overlookmotel commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12206?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#12206** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12206?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12206?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#12205** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12205?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#12204** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12204?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#12203** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12203?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=12206).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Dunqing commented:
### Merge activity

* **Jul 11, 12:55 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
fix(allocator)!: remove `vec!` macro (#12206)

Remove `vec!` macro from `oxc_allocator`. This is in the `vec2::raw_vec` module, but is accidentally exposed due to the `#[macro_export]` attribute.

This macro creates a `bumpalo::collections::Vec`, instead of our own `Vec` type, which is confusing.

We could add a `vec!` macro which creates our own `Vec` type, but that's not completely trivial (how to handle cloning `item` in `vec![item; 3]`? `Clone` or `CloneIn`?). So for now, just remove this erroneous macro entirely.
